Build failure: Kodi-inputstream-ffmpegdirect fails to build on unstable

### Steps To Reproduce

Steps to reproduce the behavior:
1. Build kodi-inputstream-ffmpegdirect on the unstable channel

### Build log

```  /nix/store/07r4wm10bza4szgxicck8hbc59gxbxdp-nixos-system-nixos-24.05pre574351.612f97239e2c.drv
building '/nix/store/dgpx4p7q2s80knpgbfsb41gz5ldk4qc9-kodi-inputstream-ffmpegdirect-20.5.0.drv'...
Running phase: unpackPhase
unpacking source archive /nix/store/96cky9haac3xa9ig6gcl4arzrbpml7wg-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/share/doc/inputstream-ffmpegdirect -DCMAKE_INSTALL_INFODIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/xq8920m5mbd83vdlydwli7qsh67gfm5v-gcc-wrapper-13.2.0/bin/strip -DCMAKE_RANLIB=/nix/store/xq8920m5mbd83vdlydwli7qsh67gfm5v-gcc-wrapper-13.2.0/bin/ranlib -DCMAKE_AR=/nix/store/xq8920m5mbd83vdlydwli7qsh67gfm5v-gcc-wrapper-13.2.0/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/qf75w06qc7yk96kypnrdyswgc6zl2p7y-kodi-inputstream-ffmpegdirect-20.5.0 -DOVERRIDE_PATHS=1 
-- The C compiler identification is GNU 13.2.0
-- The CXX compiler identification is GNU 13.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/xq8920m5mbd83vdlydwli7qsh67gfm5v-gcc-wrapper-13.2.0/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/xq8920m5mbd83vdlydwli7qsh67gfm5v-gcc-wrapper-13.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Checking to see if CXX compiler accepts flag -flto
-- Checking to see if CXX compiler accepts flag -flto - yes
-- FFmpeg 4.0 not found, falling back to internal build
-- Found ZLIB: /nix/store/mrzfvc2gjy2mia0k4vrb95f2iff6xcxm-zlib-1.3/lib/libz.so (found version "1.3")  
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE) 
-- Found BZip2: /nix/store/3vkcrwqal8sbvdc1gdi3pcwbqiszv38x-bzip2-1.0.8-dev/include  
-- FFMPEGDIRECT_VERSION=20.5.0
-- FFMPEGDIRECT_VERSION=20.5.0
-- Addon dependency check ...
--  - Added API usage definition: ADDON_GLOBAL_VERSION_MAIN_USED (Version: "2.0.2", Min. Version: "2.0.0")
--  - Added API usage definition: ADDON_GLOBAL_VERSION_GENERAL_USED (Version: "1.0.5", Min. Version: "1.0.4")
--  - Added API usage definition: ADDON_GLOBAL_VERSION_FILESYSTEM_USED (Version: "1.1.8", Min. Version: "1.1.7")
--  - Added API usage definition: ADDON_GLOBAL_VERSION_NETWORK_USED (Version: "1.0.4", Min. Version: "1.0.0")
--  - Added API usage definition: ADDON_GLOBAL_VERSION_TOOLS_USED (Version: "1.0.4", Min. Version: "1.0.0")
--  - Added API usage definition: ADDON_INSTANCE_VERSION_INPUTSTREAM_USED (Version: "3.2.0", Min. Version: "3.2.0")
-- Configuring done (3.4s)
-- Generating done (0.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
build flags: -j12 SHELL=/nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21/bin/bash
[ 25%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/FFmpegCatchupStream.cpp.o
[ 25%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/FFmpegLog.cpp.o
[ 25%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/FFmpegStream.cpp.o
[ 25%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/StreamManager.cpp.o
[ 31%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/CurlCatchupInput.cpp.o
[ 37%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/TimeshiftBuffer.cpp.o
[ 43%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/CurlInput.cpp.o
[ 50%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/DemuxStream.cpp.o
[ 56%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/TimeshiftSegment.cpp.o
[ 62%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/url/URL.cpp.o
[ 68%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/TimeshiftStream.cpp.o
[ 75%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/url/UrlOptions.cpp.o
In file included from /build/source/src/stream/FFmpegStream.h:10,
                 from /build/source/src/stream/FFmpegStream.cpp:8:
/build/source/src/stream/../utils/HttpProxy.h:18:40: error: 'uint16_t' has not been declared
   18 |     HttpProxy(const std::string& host, uint16_t port, const std::string& user, const std::string& password)
      |                                        ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:24:5: error: 'uint16_t' does not name a type
   24 |     uint16_t GetProxyPort() const { return m_port; }
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:11:1: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   10 | #include <string>
  +++ |+#include <cstdint>
   11 | 
/build/source/src/stream/../utils/HttpProxy.h:25:23: error: 'uint16_t' has not been declared
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                       ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: error: 'uint16_t' does not name a type
   35 |     uint16_t m_port;
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/build/source/src/stream/../utils/HttpProxy.h: In constructor 'ffmpegdirect::HttpProxy::HttpProxy(const std::string&, int, const std::string&, const std::string&)':
/build/source/src/stream/../utils/HttpProxy.h:19:23: error: class 'ffmpegdirect::HttpProxy' does not have any field named 'm_port'
   19 |       : m_host(host), m_port(port), m_user(user), m_password(password) {};
      |                       ^~~~~~
/build/source/src/stream/../utils/HttpProxy.h: In member function 'void ffmpegdirect::HttpProxy::SetProxyPort(int)':
/build/source/src/stream/../utils/HttpProxy.h:25:41: error: 'm_port' was not declared in this scope; did you mean 'm_host'?
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                                         ^~~~~~
      |                                         m_host
In file included from /build/source/src/stream/FFmpegStream.h:10,
                 from /build/source/src/stream/FFmpegCatchupStream.h:10,
                 from /build/source/src/stream/FFmpegCatchupStream.cpp:8:
/build/source/src/stream/../utils/HttpProxy.h:18:40: error: 'uint16_t' has not been declared
   18 |     HttpProxy(const std::string& host, uint16_t port, const std::string& user, const std::string& password)
      |                                        ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:24:5: error: 'uint16_t' does not name a type
   24 |     uint16_t GetProxyPort() const { return m_port; }
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:11:1: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   10 | #include <string>
  +++ |+#include <cstdint>
   11 | 
/build/source/src/stream/../utils/HttpProxy.h:25:23: error: 'uint16_t' has not been declared
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                       ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: error: 'uint16_t' does not name a type
   35 |     uint16_t m_port;
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/build/source/src/stream/../utils/HttpProxy.h: In constructor 'ffmpegdirect::HttpProxy::HttpProxy(const std::string&, int, const std::string&, const std::string&)':
/build/source/src/stream/../utils/HttpProxy.h:19:23: error: class 'ffmpegdirect::HttpProxy' does not have any field named 'm_port'
   19 |       : m_host(host), m_port(port), m_user(user), m_password(password) {};
      |                       ^~~~~~
/build/source/src/stream/../utils/HttpProxy.h: In member function 'void ffmpegdirect::HttpProxy::SetProxyPort(int)':
/build/source/src/stream/../utils/HttpProxy.h:25:41: error: 'm_port' was not declared in this scope; did you mean 'm_host'?
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                                         ^~~~~~
      |                                         m_host
In file included from /build/source/src/stream/TimeshiftStream.h:10,
                 from /build/source/src/stream/TimeshiftStream.cpp:8:
/build/source/src/stream/../utils/HttpProxy.h:18:40: error: 'uint16_t' has not been declared
   18 |     HttpProxy(const std::string& host, uint16_t port, const std::string& user, const std::string& password)
      |                                        ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:24:5: error: 'uint16_t' does not name a type
   24 |     uint16_t GetProxyPort() const { return m_port; }
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:11:1: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   10 | #include <string>
  +++ |+#include <cstdint>
   11 | 
/build/source/src/stream/../utils/HttpProxy.h:25:23: error: 'uint16_t' has not been declared
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                       ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: error: 'uint16_t' does not name a type
   35 |     uint16_t m_port;
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/build/source/src/stream/../utils/HttpProxy.h: In constructor 'ffmpegdirect::HttpProxy::HttpProxy(const std::string&, int, const std::string&, const std::string&)':
/build/source/src/stream/../utils/HttpProxy.h:19:23: error: class 'ffmpegdirect::HttpProxy' does not have any field named 'm_port'
   19 |       : m_host(host), m_port(port), m_user(user), m_password(password) {};
      |                       ^~~~~~
/build/source/src/stream/../utils/HttpProxy.h: In member function 'void ffmpegdirect::HttpProxy::SetProxyPort(int)':
/build/source/src/stream/../utils/HttpProxy.h:25:41: error: 'm_port' was not declared in this scope; did you mean 'm_host'?
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                                         ^~~~~~
      |                                         m_host
In file included from /build/source/src/stream/FFmpegStream.h:10,
                 from /build/source/src/StreamManager.h:10,
                 from /build/source/src/StreamManager.cpp:8:
/build/source/src/stream/../utils/HttpProxy.h:18:40: error: 'uint16_t' has not been declared
   18 |     HttpProxy(const std::string& host, uint16_t port, const std::string& user, const std::string& password)
      |                                        ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:24:5: error: 'uint16_t' does not name a type
   24 |     uint16_t GetProxyPort() const { return m_port; }
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:11:1: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
   10 | #include <string>
  +++ |+#include <cstdint>
   11 | 
/build/source/src/stream/../utils/HttpProxy.h:25:23: error: 'uint16_t' has not been declared
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                       ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: error: 'uint16_t' does not name a type
   35 |     uint16_t m_port;
      |     ^~~~~~~~
/build/source/src/stream/../utils/HttpProxy.h:35:5: note: 'uint16_t' is defined in header '<cstdint>'; did you forget to '#include <cstdint>'?
/build/source/src/stream/../utils/HttpProxy.h: In constructor 'ffmpegdirect::HttpProxy::HttpProxy(const std::string&, int, const std::string&, const std::string&)':
/build/source/src/stream/../utils/HttpProxy.h:19:23: error: class 'ffmpegdirect::HttpProxy' does not have any field named 'm_port'
   19 |       : m_host(host), m_port(port), m_user(user), m_password(password) {};
      |                       ^~~~~~
/build/source/src/stream/../utils/HttpProxy.h: In member function 'void ffmpegdirect::HttpProxy::SetProxyPort(int)':
/build/source/src/stream/../utils/HttpProxy.h:25:41: error: 'm_port' was not declared in this scope; did you mean 'm_host'?
   25 |     void SetProxyPort(uint16_t value) { m_port = value; }
      |                                         ^~~~~~
      |                                         m_host
In file included from /build/source/src/stream/DemuxStream.cpp:8:
/build/source/src/stream/DemuxStream.h:36:7: warning: 'ffmpegdirect::DemuxStream' declared with greater visibility than the type of its field 'ffmpegdirect::DemuxStream::cryptoSession' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   36 | class DemuxStream
      |       ^~~~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:13:
/build/source/src/stream/DemuxStream.h: At global scope:
/build/source/src/stream/DemuxStream.h:36:7: warning: 'ffmpegdirect::DemuxStream' declared with greater visibility than the type of its field 'ffmpegdirect::DemuxStream::cryptoSession' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   36 | class DemuxStream
      |       ^~~~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:13,
                 from /build/source/src/stream/TimeshiftStream.h:12:
/build/source/src/stream/DemuxStream.h: At global scope:
/build/source/src/stream/DemuxStream.h:36:7: warning: 'ffmpegdirect::DemuxStream' declared with greater visibility than the type of its field 'ffmpegdirect::DemuxStream::cryptoSession' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   36 | class DemuxStream
      |       ^~~~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:14:
/build/source/src/stream/CurlInput.h:23:7: warning: 'ffmpegdirect::CurlInput' declared with greater visibility than the type of its field 'ffmpegdirect::CurlInput::m_pFile' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   23 | class CurlInput
      |       ^~~~~~~~~
In file included from /build/source/src/stream/CurlInput.cpp:8:
/build/source/src/stream/CurlInput.h:23:7: warning: 'ffmpegdirect::CurlInput' declared with greater visibility than the type of its field 'ffmpegdirect::CurlInput::m_pFile' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   23 | class CurlInput
      |       ^~~~~~~~~
In file included from /build/source/src/stream/CurlCatchupInput.h:10,
                 from /build/source/src/stream/CurlCatchupInput.cpp:8:
/build/source/src/stream/CurlInput.h:23:7: warning: 'ffmpegdirect::CurlInput' declared with greater visibility than the type of its field 'ffmpegdirect::CurlInput::m_pFile' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   23 | class CurlInput
      |       ^~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:14:
/build/source/src/stream/CurlInput.h:23:7: warning: 'ffmpegdirect::CurlInput' declared with greater visibility than the type of its field 'ffmpegdirect::CurlInput::m_pFile' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   23 | class CurlInput
      |       ^~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:13:
/build/source/src/stream/DemuxStream.h: At global scope:
/build/source/src/stream/DemuxStream.h:36:7: warning: 'ffmpegdirect::DemuxStream' declared with greater visibility than the type of its field 'ffmpegdirect::DemuxStream::cryptoSession' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   36 | class DemuxStream
      |       ^~~~~~~~~~~
In file included from /build/source/src/stream/TimeshiftSegment.cpp:8:
/build/source/src/stream/TimeshiftSegment.h:27:7: warning: 'ffmpegdirect::TimeshiftSegment' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftSegment::m_fileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   27 | class TimeshiftSegment
      |       ^~~~~~~~~~~~~~~~
In file included from /build/source/src/stream/TimeshiftBuffer.h:11,
                 from /build/source/src/stream/TimeshiftBuffer.cpp:8:
/build/source/src/stream/TimeshiftSegment.h:27:7: warning: 'ffmpegdirect::TimeshiftSegment' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftSegment::m_fileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   27 | class TimeshiftSegment
      |       ^~~~~~~~~~~~~~~~
/build/source/src/stream/TimeshiftBuffer.h:37:7: warning: 'ffmpegdirect::TimeshiftBuffer' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftBuffer::m_segmentIndexFileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   37 | class TimeshiftBuffer
      |       ^~~~~~~~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:14:
/build/source/src/stream/CurlInput.h:23:7: warning: 'ffmpegdirect::CurlInput' declared with greater visibility than the type of its field 'ffmpegdirect::CurlInput::m_pFile' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   23 | class CurlInput
      |       ^~~~~~~~~
In file included from /build/source/src/stream/TimeshiftBuffer.h:11,
                 from /build/source/src/stream/TimeshiftStream.h:13:
/build/source/src/stream/TimeshiftSegment.h:27:7: warning: 'ffmpegdirect::TimeshiftSegment' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftSegment::m_fileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   27 | class TimeshiftSegment
      |       ^~~~~~~~~~~~~~~~
/build/source/src/stream/TimeshiftBuffer.h:37:7: warning: 'ffmpegdirect::TimeshiftBuffer' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftBuffer::m_segmentIndexFileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   37 | class TimeshiftBuffer
      |       ^~~~~~~~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:13:
/build/source/src/stream/DemuxStream.h: At global scope:
/build/source/src/stream/DemuxStream.h:36:7: warning: 'ffmpegdirect::DemuxStream' declared with greater visibility than the type of its field 'ffmpegdirect::DemuxStream::cryptoSession' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   36 | class DemuxStream
      |       ^~~~~~~~~~~
In file included from /build/source/src/stream/FFmpegStream.h:14:
/build/source/src/stream/CurlInput.h:23:7: warning: 'ffmpegdirect::CurlInput' declared with greater visibility than the type of its field 'ffmpegdirect::CurlInput::m_pFile' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   23 | class CurlInput
      |       ^~~~~~~~~
/build/source/src/stream/FFmpegStream.cpp: In member function 'AVDictionary* ffmpegdirect::FFmpegStream::GetFFMpegOptionsFromInput()':
/build/source/src/stream/FFmpegStream.cpp:2195:39: error: 'class ffmpegdirect::HttpProxy' has no member named 'GetProxyPort'; did you mean 'SetProxyPort'?
 2195 |     const uint16_t port = m_httpProxy.GetProxyPort();
      |                                       ^~~~~~~~~~~~
      |                                       SetProxyPort
In file included from /build/source/src/stream/TimeshiftBuffer.h:11,
                 from /build/source/src/stream/TimeshiftStream.h:13,
                 from /build/source/src/StreamManager.cpp:11:
/build/source/src/stream/TimeshiftSegment.h:27:7: warning: 'ffmpegdirect::TimeshiftSegment' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftSegment::m_fileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   27 | class TimeshiftSegment
      |       ^~~~~~~~~~~~~~~~
/build/source/src/stream/TimeshiftBuffer.h:37:7: warning: 'ffmpegdirect::TimeshiftBuffer' declared with greater visibility than the type of its field 'ffmpegdirect::TimeshiftBuffer::m_segmentIndexFileHandle' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wattributes-Wattributes8;;]
   37 | class TimeshiftBuffer
      |       ^~~~~~~~~~~~~~~
/build/source/src/StreamManager.cpp: In member function 'virtual bool InputStreamFFmpegDirect::Open(const kodi::addon::InputstreamProperty&)':
/build/source/src/StreamManager.cpp:202:84: error: 'class ffmpegdirect::HttpProxy' has no member named 'GetProxyPort'; did you mean 'SetProxyPort'?
  202 |     kodi::Log(ADDON_LOG_INFO, "HttpProxy port set: %d", static_cast<int>(httpProxy.GetProxyPort()));
      |                                                                                    ^~~~~~~~~~~~
      |                                                                                    SetProxyPort
[ 81%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/url/Variant.cpp.o
[ 87%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/utils/DiskUtils.cpp.o
[ 93%] Building CXX object CMakeFiles/inputstream.ffmpegdirect.dir/src/utils/FilenameUtils.cpp.o
make[2]: *** [CMakeFiles/inputstream.ffmpegdirect.dir/build.make:132: CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/FFmpegStream.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/inputstream.ffmpegdirect.dir/build.make:104: CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/FFmpegCatchupStream.cpp.o] Error 1
make[2]: *** [CMakeFiles/inputstream.ffmpegdirect.dir/build.make:202: CMakeFiles/inputstream.ffmpegdirect.dir/src/stream/TimeshiftStream.cpp.o] Error 1
make[2]: *** [CMakeFiles/inputstream.ffmpegdirect.dir/build.make:76: CMakeFiles/inputstream.ffmpegdirect.dir/src/StreamManager.cpp.o] Error 1
In file included from /build/source/src/utils/DiskUtils.cpp:8:
/build/source/src/utils/DiskUtils.h:17:61: error: 'uint64_t' has not been declared
   17 |     static bool GetFreeDiskSpaceMB(const std::string& path, uint64_t& freeMB);
      |                                                             ^~~~~~~~
/build/source/src/utils/DiskUtils.cpp:18:6: error: no declaration matches 'bool ffmpegdirect::DiskUtils::GetFreeDiskSpaceMB(const std::string&, uint64_t&)'
   18 | bool DiskUtils::GetFreeDiskSpaceMB(const std::string& path, uint64_t& freeMB)
      |      ^~~~~~~~~
/build/source/src/utils/DiskUtils.h:17:17: note: candidate is: 'static bool ffmpegdirect::DiskUtils::GetFreeDiskSpaceMB(const std::string&, int&)'
   17 |     static bool GetFreeDiskSpaceMB(const std::string& path, uint64_t& freeMB);
      |                 ^~~~~~~~~~~~~~~~~~
/build/source/src/utils/DiskUtils.h:14:9: note: 'class ffmpegdirect::DiskUtils' defined here
   14 |   class DiskUtils
      |         ^~~~~~~~~
make[2]: *** [CMakeFiles/inputstream.ffmpegdirect.dir/build.make:258: CMakeFiles/inputstream.ffmpegdirect.dir/src/utils/DiskUtils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:110: CMakeFiles/inputstream.ffmpegdirect.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
error: builder for '/nix/store/dgpx4p7q2s80knpgbfsb41gz5ldk4qc9-kodi-inputstream-ffmpegdirect-20.5.0.drv' failed with exit code 2

```
log here if short otherwise a link to a gist
`

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
output here
`` - system: `"x86_64-linux"`
 - host os: `Linux 6.7.0, NixOS, 23.11 (Tapir), 23.11.3019.8bf65f17d807`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/video/kodi/addons/inputstream-ffmpegdirect/default.nix
+++ b/pkgs/applications/video/kodi/addons/inputstream-ffmpegdirect/default.nix
@@ -3,13 +3,13 @@
 buildKodiBinaryAddon rec {
   pname = "inputstream-ffmpegdirect";
   namespace = "inputstream.ffmpegdirect";
-  version = "20.5.0";
+  version = "unstable-20.5.0";
 
   src = fetchFromGitHub {
     owner = "xbmc";
     repo = "inputstream.ffmpegdirect";
-    rev = "${version}-${rel}";
-    sha256 = "sha256-2Xa5q+o/AYDwG+JYeXTUfMTJk/kln16J8KRcXILNE6c=";
+    rev = rel;
+    sha256 = "sha256-+u28Wzp2TonL5jaa5WJUr9igR6KiaxizZAX9jqqBUns=";
   };
 
   extraBuildInputs = [ bzip2 zlib kodi.ffmpeg ];

